### DISCUSSION TITLE
Google Home and Matter integration question

### DISCUSSION THREAD

**Discussion by ivanvach** [arendst/Tasmota#20646] [Q&A] [answered]
I did adopt Tasmota driven ESP32 board as Matter enabled device in my Google Home. The question is how did that happen provided that ESP32 doesn't support Thread. Did it happen over UDP or TCP or something else.

**Comment by s-hadinger** [comment]
Yes, it's Matter over Wifi. Thread is a separate radio protocol aimed at replacing Zigbee. Matter supports both Wifi and Thread.

**Comment by ivanvach** [comment]
Thanks. Still curious does it use UDP, TCP or something else like PASE (Pairing and Smart Energy) or CASE (CHIP Authentication and Secure Establishment). Thanks.

**Reply by s-hadinger** [reply, marked as answer]
Only UDP.

PASE (Passcode-Authenticated Session Establishment) and CASE (Certificate Authenticated Session Establishment) are just 2 protocols used for authentication.

**Reply by ivanvach** [reply]
@s-hadinger Thanks. What if the Tasmota Device comes from the factory with Tasmota pre-flashed. Can we skip the manual entry of WiFi credentials and just provision it with Matter provided that QR is on the box? UDP won't work in this case. Maybe Bluetooth can be utilized ?

**Reply by s-hadinger** [reply]
In theory yes. But it would require to implement the BLE part of Matter. I will not do it, so unless someone is ready to spend long hours on it, it will not happen 

There are off the shelf solutions from Espressif for this

**Reply by Jason2866** [reply]
There is already a PoC for onboarding Tasmota via BLE. No need to do it via Matter. 
https://github.com/Staars/improv_wifi_ble

To be clear this will not released from Tasmota crew since it needs paid Developer Accounts to bring the APPs in the different mobile phone APP stores.
To be more general there will no kind of any official Tasmota APPs in mobile phones APP stores.
